src/filesystem_helper.cpp

```diff
@@ -306,6 +306,22 @@ path temp_directory_path()
   return path(temp_path);
 }
 
+path create_temp_directory(const std::string & base_name, const path & parent)
+{
+  auto template_path = base_name + ".XXXXXX";
+
+#ifdef _WIN32
+  _mktemp_s(&template_path[0], template_path.size() + 1);
```

Comment:
@emersonknapp consider using [`std::string::c_str()`](https://en.cppreference.com/w/cpp/string/basic_string/c_str) instead.

Reply:
In these APIs, the input `char *` must be mutable, because the `XXXXXX` is overwritten by the API. `.c_str()` (or `.data()` in C++11) return a `const char *` that can't be used here.

Reply:
Hmm, good point. I'd rather not make assumptions about `std::string` underlying structure either (i.e. using `&var[0]`). I guess we're left with primitive char arrays or `std::vector<char>` (using `std::vector::data`).

Reply:
As mentioned below by @clalancette, `std::basic_string` does guarantee contiguous storage such that `&var[0]` is well-defined. Nevermind my last comment.